### PR TITLE
BatchNormInference specification refactoring

### PR DESCRIPTION
**Pull request by ggalieroc**
BatchNormInference specification refactored.

Details:

- Re-write Short Description
- Added Detailed Description with mathematical formulas
- Re-write attribute description and range of values
- Re-write inputs and outputs description
- Added type section
- Added example with 2D input tensor

Ticket: 46111

### CODE REVIEW

File: docs/ops/normalization/BatchNormInference_1.md [hunk 1]
```diff
@@ -4,39 +4,33 @@
 
 **Category**: *Normalization*
 
-**Short description**: *BatchNormInference* layer normalizes a `input` tensor by `mean` and `variance`, and applies a scale (`gamma`) to it, as well as an offset (`beta`).
+**Short description**: *BatchNormInference* performs Batch Normalizing Transform along a data batch. [Reference](https://arxiv.org/abs/1502.03167v2)
```
Comment (lazarevevgeny) [inline docs/ops/normalization/BatchNormInference_1.md:7]: Let's change to:
```
**Short description**: *BatchNormInference* performs Batch Normalization operation described in the [Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift](https://arxiv.org/abs/1502.03167v2) article.
```

I don't agree to use the word "Transform" because it is not a transformation but a mathematical operation.

File: docs/ops/normalization/BatchNormInference_1.md [hunk 2]
```diff
@@ -4,39 +4,33 @@
...
-* **2**: `gamma` - gamma scaling for normalized value. A 1D tensor of type T with the same span as input's channel axis. **Required.**
-* **3**: `beta` - bias added to the scaled normalized value. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
-* **4**: `mean` - value for mean normalization. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
-* **5**: `variance` - value for variance normalization. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
+* Normalize each activation \f$x^{(k)}\f$ by means of the mean and variance.
+\f[
+   \hat{x}^{(k)}=\frac{x^{(k)} - E[x^{(k)}]}{\sqrt{Var(x^{(k)}) + \epsilon}}
+\f]
+where \f$E[x^{(k)}]\f$ and \f$Var(x^{(k)})\f$ are the mean and variance, calculated along the channel axis of the `data` input, and correspond to `mean` and `variance` inputs respectively. Additionally, \f$\epsilon\f$ is a value added to the variance for numerical stability and corresponds to `epsilon` attribute.
```
Comment (lazarevevgeny) [inline docs/ops/normalization/BatchNormInference_1.md:17]: May be change "calculated along the channel axis" to "calculated per the channel axis".

File: docs/ops/normalization/BatchNormInference_1.md [hunk 3]
```diff
@@ -4,39 +4,33 @@
...
+\f]
+where \f$\gamma^{(k)}\f$ and \f$\beta^{(k)}\f$ are learnable parameters, calculated along the channel axis, and correspond to `gamma` and `beta` inputs.
 
-* **1**: The result of normalization. A tensor of the same type and shape with 1st input tensor.
-
-**Types**
+**Mathematical Formulation**
 
-* *T*: any numeric type.
+Let x be a *d*-dimensional input, \f$x=(x_{1}\dotsc x_{d})\f$. Since normalization is applied to each activation \f$E[x^{(k)}]\f$, we may focus on a particular activation and ommit k.
```
Comment (lazarevevgeny) [inline docs/ops/normalization/BatchNormInference_1.md:27]: wrap "x" with back ticks

File: docs/ops/normalization/BatchNormInference_1.md [hunk 4]
```diff
@@ -45,22 +39,81 @@
...
     \f[
     o_{i} \leftarrow \gamma\hat{b_{i}} + \beta = BN_{\gamma ,\beta } ( b_{i} )
     \f]
 
-**Example**
+**Attributes**:
+
+* *epsilon*
+  * **Description**: *epsilon* is a constant added to the variance for numerical stability
+  * **Range of values**: a positive non-zero floating-point number
```
Comment (lazarevevgeny) [inline docs/ops/normalization/BatchNormInference_1.md:61]: 'positive' already means 'non-zero'. So change to "a positive floating-point number"

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]
Please, send me in the email the rendered version of the document.

File: docs/ops/normalization/BatchNormInference_1.md (hunk 1 above)
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: The reason I used this term to align with the way the paper names the algorithm. But it is true that causes confusion, so I have removed it.

File: docs/ops/normalization/BatchNormInference_1.md (hunk 2 above)
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Done

File: docs/ops/normalization/BatchNormInference_1.md (hunk 3 above)
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Done

File: docs/ops/normalization/BatchNormInference_1.md (hunk 4 above)
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Done

File: docs/ops/normalization/BatchNormInference_5.md [hunk 5]
```diff
@@ -1,42 +1,36 @@
...
+where \f$\gamma^{(k)}\f$ and \f$\beta^{(k)}\f$ are learnable parameters, calculated along the channel axis, and correspond to `gamma` and `beta` inputs.
 
-**Types**
+**Mathematical Formulation**
 
-* *T*: any numeric type.
+Let `x` be a *d*-dimensional input, \f$x=(x_{1}\dotsc x_{d})\f$. Since normalization is applied to each activation \f$E[x^{(k)}]\f$, we may focus on a particular activation and ommit k.
 
-**Mathematical Formulation**
+Considering a mini-batch \f$\mathcal{B}\f$ of m values for a particular activation. *BatchNormInference* performs Batch Normalizing Transform algorithm, as follows:
```
Comment (lazarevevgeny) [inline docs/ops/normalization/BatchNormInference_5.md:29]: Still see "Transform algorithm". Please, change.

**Review by lazarevevgeny — APPROVED** [review]
From technical point of view it looks good. But, please, ask technical writers for review since there are a lot of changes

File: docs/ops/normalization/BatchNormInference_5.md (hunk 5 above)
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_5.md]: Removed

File: docs/ops/normalization/BatchNormInference_1.md [hunk 6]
```diff
@@ -4,39 +4,33 @@
...
-* **1**: The result of normalization. A tensor of the same type and shape with 1st input tensor.
-
-**Types**
+**Mathematical Formulation**
 
-* *T*: any numeric type.
+Let `x` be a *d*-dimensional input, \f$x=(x_{1}\dotsc x_{d})\f$. Since normalization is applied to each activation \f$E[x^{(k)}]\f$, we may focus on a particular activation and ommit k.
 
-**Mathematical Formulation**
+Considering a mini-batch \f$\mathcal{B}\f$ of m values for a particular activation. *BatchNormInference* performs Batch Normalization algorithm, as follows:
```
Comment (tsavina) [inline docs/ops/normalization/BatchNormInference_1.md:29]: Considering a mini-batch \f$\mathcal{B}\f$ of m values for a particular activation. *BatchNormInference* performs Batch Normalization algorithm, as follows: --->
For a particular activation, consider a mini-batch \f$\mathcal{B}\f$ of m values. *BatchNormInference* performs Batch Normalization algorithm as follows:

File: docs/ops/normalization/BatchNormInference_1.md [hunk 7]
```diff
@@ -4,39 +4,33 @@
...
-* *epsilon*
-  * **Description**: *epsilon* is the number to be added to the variance to avoid division by zero when normalizing a value. For example, *epsilon* equal to 0.001 means that 0.001 is added to the variance.
-  * **Range of values**: a positive floating-point number
-  * **Type**: `float`
-  * **Default value**: None
-  * **Required**: *yes*
+*BatchNormInference* performs the following operations on a given data batch input tensor `data`:
 
-**Inputs**
+* Normalize each activation \f$x^{(k)}\f$ by means of the mean and variance.
```
[the remark this answers is not in the document]
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Done

File: docs/ops/normalization/BatchNormInference_1.md [hunk 8]
```diff
@@ -4,39 +4,33 @@
...
+\f[
+   \hat{x}^{(k)}=\frac{x^{(k)} - E[x^{(k)}]}{\sqrt{Var(x^{(k)}) + \epsilon}}
+\f]
+where \f$E[x^{(k)}]\f$ and \f$Var(x^{(k)})\f$ are the mean and variance, calculated per the channel axis of `data` input, and correspond to `mean` and `variance` inputs respectively. Additionally, \f$\epsilon\f$ is a value added to the variance for numerical stability and corresponds to `epsilon` attribute.
```
[the remark this answers is not in the document]
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Done

File: docs/ops/normalization/BatchNormInference_1.md [hunk 9]
```diff
@@ -4,39 +4,33 @@
...
 
-* **1**: `input` - input tensor with data for normalization. At least a 2D tensor of type T, the second dimension represents the channel axis and must have a span of at least 1. **Required.**
-* **2**: `gamma` - gamma scaling for normalized value. A 1D tensor of type T with the same span as input's channel axis. **Required.**
-* **3**: `beta` - bias added to the scaled normalized value. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
-* **4**: `mean` - value for mean normalization. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
-* **5**: `variance` - value for variance normalization. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
+* Linear transformation of each normalized activation based on `gamma` and `beta` input, representing the scaling factor and shift respectively.
```
[the remark this answers is not in the document]
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Done

File: docs/ops/normalization/BatchNormInference_1.md [hunk 10]
```diff
@@ -4,39 +4,33 @@
...
 
-**Outputs**
-
-* **1**: The result of normalization. A tensor of the same type and shape with 1st input tensor.
-
-**Types**
+**Mathematical Formulation**
 
-* *T*: any numeric type.
+Let `x` be a *d*-dimensional input, \f$x=(x_{1}\dotsc x_{d})\f$. Since normalization is applied to each activation \f$E[x^{(k)}]\f$, we may focus on a particular activation and ommit k.
```
[the remark this answers is not in the document]
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Definitely agree. Done

File: docs/ops/normalization/BatchNormInference_1.md (hunk 6 above)
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Done

File: docs/ops/normalization/BatchNormInference_1.md [hunk 11]
```diff
@@ -4,39 +4,33 @@
...
-* **1**: `input` - input tensor with data for normalization. At least a 2D tensor of type T, the second dimension represents the channel axis and must have a span of at least 1. **Required.**
-* **2**: `gamma` - gamma scaling for normalized value. A 1D tensor of type T with the same span as input's channel axis. **Required.**
-* **3**: `beta` - bias added to the scaled normalized value. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
-* **4**: `mean` - value for mean normalization. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
-* **5**: `variance` - value for variance normalization. A 1D tensor of type T with the same span as input's channel axis.. **Required.**
+* Linear transformation of each normalized activation based on `gamma` and `beta` input, representing the scaling factor and shift respectively.
+\f[
+   \hat{y}^{(k)}=\gamma^{(k)}\hat{x}^{(k)} + \beta^{(k)}
+\f]
+where \f$\gamma^{(k)}\f$ and \f$\beta^{(k)}\f$ are learnable parameters, calculated along the channel axis, and correspond to `gamma` and `beta` inputs.
```
[the remark this answers is not in the document]
Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Definitely agree. Done

File: docs/ops/normalization/BatchNormInference_1.md [hunk 12]
```diff
@@ -4,39 +4,33 @@
...
-* *epsilon*
-  * **Description**: *epsilon* is the number to be added to the variance to avoid division by zero when normalizing a value. For example, *epsilon* equal to 0.001 means that 0.001 is added to the variance.
-  * **Range of values**: a positive floating-point number
-  * **Type**: `float`
-  * **Default value**: None
-  * **Required**: *yes*
+*BatchNormInference* performs the following operations on a given data batch input tensor `data`:
 
-**Inputs**
+* Normalize each activation \f$x^{(k)}\f$ by the mean and variance.
```
Comment (tsavina) [inline docs/ops/normalization/BatchNormInference_1.md:13]: Normalize each activation --> Normalizes each activation \f$x^{(k)}\f$ by the mean and variance.
Can we make this in 3d person so it's more consistent? Example: Performs operations: - Normalizes... -Performs linear transformation

Comment (ggalieroc) [reply docs/ops/normalization/BatchNormInference_1.md]: Absolutely, I am sorry I missed it in the previous comment

**Review by tsavina — APPROVED** [review]
